README.md

```diff
@@ -23,368 +23,298 @@ Madara is a powerful Starknet client written in Rust.
-
-   - **Release** (recommended build mode):
-
-     ```bash
-     cargo build --release
-     ```
-
-   - **Production** (recommended for production performance):
-
-     ```bash
-     cargo build --profile=production
-     ```
-
-3. **Run Madara**
-
-   Start the Madara client with a basic set of arguments depending on your chosen mode:
-
-   **Full Node**
-
-   ```bash
-   cargo run --release -- \
-     --name Madara \
-     --full \
-     --base-path /var/lib/madara \
-     --network mainnet \
-     --l1-endpoint ${ETHEREUM_API_URL}
-   ```
-
-   **Sequencer**
+[⬅️ back to top](#-madara-starknet-client)
 
-   ```bash
-   cargo run --release -- \
-     --name Madara \
-     --sequencer \
-     --base-path /var/lib/madara \
-     --preset test \
-     --l1-endpoint ${ETHEREUM_API_URL}
-   ```
-
-   **Devnet**
-
-   ```bash
-   cargo run --release -- \
-     --name Madara \
-     --devnet \
-     --base-path /var/lib/madara \
-     --preset test
-   ```
-
-   > ℹ️ **Info:** We recommend you to head to the [Configuration](https://docs.madara.build/fundamentals/configuration)
-   > section to customize your node parameters.
-   > ℹ️ **Info:** If you don't have an L1 endpoint URL, we recommend you refer to the relevant
-   > section to obtain one.
-
-### Run with Docker
-
-1. **Install Docker**
-
-   Ensure you have Docker installed on your machine. Once you have Docker installed, you can run Madara using the available Docker images.
-
-   ```bash
-   docker run -d \
-     --name Madara \
-     --full
-     -p 9944:9944 \
-     -v /var/lib/madara:/var/lib/madara \
-     madara:latest \
-     --base-path /var/lib/madara \
-     --network mainnet \
-     --l1-endpoint ${ETHEREUM_API_URL}
-   ```
-
-   > ℹ️ **Info:** This is a default configuration for a Full Node on Starknet mainnet.
-   > For more information on possible configurations, please visit the
-   > [Configuration](https://docs.madara.build/fundamentals/configuration) section.
-   > ⚠️ **Warning:** Make sure to change the volume `-v` of your container if you change the `--base-path`.
-   > ℹ️ **Info:** If you don't have an L1 endpoint URL, we recommend you refer to the relevant section to obtain one.
-
-2. **Check Logs**
-
-   ```bash
-   docker logs -f Madara
-   ```
+### Run from Source
 
-   > ℹ️ **Info:** Now you can head to the [Metrics](https://docs.madara.build/monitoring/grafana)
-   > section to deploy a Grafana and Prometheus dashboard.
+#### 1. Install dependencies
 
-## ⚙️ Configuration
+Ensure you have all the necessary dependencies available on your host system.
 
-### Command-Line Options
+| Dependency | Version    | Installation                                                      |
+| ---------- | ---------- | ----------------------------------------------------------------- |
+| Rust       | rustc 1.81 | `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs \| sh` |
+| Clang      | Latest     | `sudo apt-get install clang`                                      |
```

Comment:
additionally, the openSSL library is required

Reply:
Thanks!